File: Mini-Instagram/Notification.cpp
#include "Notification.h"

Notification::Notification() {}

Notification::~Notification() {}

void Notification::sendNotification(const string& sender, const string& about, const time_t& now) {
	notifications.enqueue(NotificationNode({ sender, about, now }));
}

NotificationNode Notification::processNotification() {
	NotificationNode recent{ notifications.frontValue() };
	notifications.dequeue();
	return recent;
}

void Notification::clearAllNotifications() {
	notifications.makeNull();
}

Queue<NotificationNode> Notification::getNotifications() const {
	return notifications;
}

void Notification::viewNotifications() const {
	Queue<NotificationNode> temp{ notifications };
	NotificationNode tempNotification;

	cout << "<-------------------->\n";
	while (!temp.isEmpty()) {
		tempNotification = temp.frontValue();
		temp.dequeue();
		cout << tempNotification.about << " from " << tempNotification.sender << endl;
		tm* time{ localtime(&tempNotification.timestamp) };
		cout << "Timestamp: " << put_time(time, "%Y-%m-%dT%H:%M:%S") << endl;
		cout << "<-------------------->\n";
	}
}

File: Mini-Instagram-Python/Notification.py
from copy import deepcopy
from datetime import datetime
from Queue import Queue


class NotificationNode:
    def __init__(self, sender, about, timestamp):
        self.sender = sender
        self.about = about
        self.timestamp = timestamp


class Notification:
    def __init__(self):
        self.notifications = Queue()

    def sendNotification(self, sender, about, now):
        self.notifications.enqueue(NotificationNode(sender, about, now))

    def processNotification(self):
        recent = self.notifications.frontValue()
        self.notifications.dequeue()
        return recent
    
    def clearAllNotifications(self):
        self.notifications.makeNull()

    def getNotifications(self):
        return self.notifications
    
    def viewNotifications(self):
        temp = deepcopy(self.notifications)
        print("<-------------------->")
        while not temp.isEmpty():
            notification = temp.frontValue()
            temp.dequeue()
            print(f"{notification.about} from {notification.sender}")
            timestamp = notification.timestamp
            time = datetime.fromtimestamp(timestamp)
            print("Timestamp: ", time.strftime("%Y-%m-%dT%H:%M:%S"))
            print("<-------------------->")